packages/lesswrong/server/collections/automatedContentEvaluations/helpers.ts

```diff
@@ -0,0 +1,278 @@
+
+DECISION RULE  
+• If any hard block fired → review.
+• Else sum soft cues.
+• Total ≥ 2 → accepted, otherwise review.
+• If you are uncertain → review.
+
+FORMATTING RULES
+• Wrap your entire reasoning in <THOUGHT> … </THOUGHT>.
+• Inside the tag, show a table:
+
+  Cue | Score
+  ----|------
+  clear structure | +1
+  … | …
+  **Total** | −1
+
+Do NOT use cues that are not listed above.
+
+EXAMPLE  
+<THOUGHT>
+The post is well-written and clear. It uses simple English, and isn't trying to sound academic. It is not exceptionally clear, so only +1.
+
+[... more cues ...]
+
+Cue | Score
+--- | ---
+well-written | +1
+**Total** | 0
+
+Total 1, newcomer, threshold 2 → review.
+</THOUGHT>
+
+EXAMPLE
+<THOUGHT>
+The post uses the term "recursion" in the title, which is a keyword that should be reviewed.
+
+Therefore, this post should be reviewed.
+</THOUGHT>
+──────────────────────────────────────────────────────────────────`;
+
+  const tools = [
+    {
+      description: "Mark a post as either accepted or requiring human review",
+      strict: true,
+      type: "function" as const,
+      name: "processPost",
+      parameters: {
+        type: "object",
+        properties: {
+          decision: {
+            type: "string",
+            enum: ["review", "accepted"],
+          },
+          reason: {
+            type: "string",
+            description: "A short explanation of the decision",
+          },
+        },
+        additionalProperties: false,
+        required: ["decision", "reason"],
+      },
+    },
+  ];
+  const markdown = dataToMarkdown(revision.html, "html");
+  const { output_text, output } = await openAIClient.responses.create({
+    model: "o4-mini",
+    instructions: rejectionSystemPrompt,
+    input: `
+<Author information>
+Author name: ${user.displayName}
+Author created at: ${user.createdAt}
+</Author information>
+
+<Post>
+${post.title}
+${markdown.slice(0, 10000)}${markdown.length > 10000 ? "... (truncated)" : ""}
+</Post>
+`,
+    tools,
+    tool_choice: "auto",
+  });
+  let toolCall = output.find((c) => c.type === "function_call");
+  if (!toolCall) {
+    const toolResponse = await openAIClient.responses.create({
+      model: "gpt-4.1",
+      instructions: rejectionSystemPrompt,
+      input: `
+Given this previous response, go ahead and call the function 'processPost' now:
+${output_text}`,
+      tools,
+      tool_choice: "required",
+    });
+    toolCall = toolResponse.output.find((c) => c.type === "function_call");
+  }
+  if (!toolCall || toolCall.name !== "processPost") {
+    // eslint-disable-next-line no-console
+    console.error("No function call 'processPost' returned");
+    return;
+  }
+  const args = JSON.parse(toolCall.arguments || "{}");
+  return {
+    cot: output_text,
+    decision: args.decision,
+    reasoning: args.reason,
+  };
+};
+
+const NO_LLM_AUTOREJECT_TEMPLATE = "No LLM (autoreject)";
```

Comment:
On second thought, we probably shouldn't be doing a lookup by name (rather than ID) because the name is editable and if someone does change it, this will just silently break.